x-pack/plugins/endpoint/common/generate_data.ts

```diff
@@ -0,0 +1,410 @@
+        action: randomChoice(FILE_OPERATIONS),
+        kind: 'alert',
+        category: 'malware',
+        id: uuid.v4(),
+        dataset: 'endpoint',
+        module: 'endpoint',
+        type: 'creation',
+      },
+      endpoint: {
+        policy: {
+          id: this.policy.id,
+        },
+      },
+      file: {
+        owner: 'SYSTEM',
+        name: 'fake_malware.exe',
+        path: 'C:/fake_malware.exe',
+        accessed: ts,
+        mtime: ts,
+        created: ts,
+        size: 3456,
+        hash: {
+          md5: 'fake file md5',
+          sha1: 'fake file sha1',
+          sha256: 'fake file sha256',
+        },
+        code_signature: {
+          trusted: false,
+          subject_name: 'bad signer',
+        },
+        malware_classifier: {
+          identifier: 'endpointpe',
+          score: 1,
+          threshold: 0.66,
+          version: '3.0.33',
+        },
+        temp_file_path: 'C:/temp/fake_malware.exe',
+      },
+      host: {
+        id: this.hostId,
+        hostname: this.hostname,
+        ip: this.ip,
+        mac: this.macAddress,
+        os: this.os,
+      },
+      process: {
+        pid: 2,
+        name: 'malware writer',
+        start: ts,
+        uptime: 0,
+        user: 'SYSTEM',
+        entity_id: entityID ? entityID : randomString(10),
+        parent: parentEntityID ? { entity_id: parentEntityID, pid: 1 } : undefined,
+        token: {
+          domain: 'NT AUTHORITY',
+          integrity_level: 16384,
+          integrity_level_name: 'system',
+          privileges: [
+            {
+              description: 'Replace a process level token',
+              enabled: false,
+              name: 'SeAssignPrimaryTokenPrivilege',
+            },
+          ],
+          sid: 'S-1-5-18',
+          type: 'tokenPrimary',
+          user: 'SYSTEM',
+        },
+        code_signature: {
+          trusted: false,
+          subject_name: 'bad signer',
+        },
+        hash: {
+          md5: 'fake md5',
+          sha1: 'fake sha1',
+          sha256: 'fake sha256',
+        },
+      },
+      dll: [
+        {
+          pe: {
+            architecture: 'x64',
+            imphash: 'c30d230b81c734e82e86e2e2fe01cd01',
+          },
+          code_signature: {
+            subject_name: 'Cybereason Inc',
+            trusted: true,
+          },
+          compile_time: 1534424710,
+          hash: {
+            md5: '1f2d082566b0fc5f2c238a5180db7451',
+            sha1: 'ca85243c0af6a6471bdaa560685c51eefd6dbc0d',
+            sha256: '8ad40c90a611d36eb8f9eb24fa04f7dbca713db383ff55a03aa0f382e92061a2',
+          },
+          malware_classifier: {
+            identifier: 'Whitelisted',
+            score: 0,
+            threshold: 0,
+            version: '3.0.0',
+          },
+          mapped_address: 5362483200,
+          mapped_size: 0,
+          path: 'C:\\Program Files\\Cybereason ActiveProbe\\AmSvc.exe',
+        },
+      ],
+    };
+  }
+
+  generateEvent(
```

Comment:
There's a couple places where `generateEvent` is called with `undefined` as the parameters, doing something like this might be more readable:
https://stackoverflow.com/questions/45416321/avoid-passing-undefined-for-optional-parameters